### PR TITLE
Resolver: switching to statusor

### PR DESCRIPTION
**Pull request by alyssawilk**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]
https://github.com/envoyproxy/envoy-mobile/issues/176

### CODE REVIEW

**Comment by alyssawilk** [comment]
@wbpcode ping?

File: source/common/network/resolver_impl.cc [hunk 1]
```diff
@@ -79,7 +80,9 @@ resolveProtoSocketAddress(const envoy::config::core::v3::SocketAddress& socket_a
   if (resolver == nullptr) {
     throwEnvoyExceptionOrPanic(fmt::format("Unknown address resolver: {}", resolver_name));
   }
-  return resolver->resolve(socket_address);
+  auto instance_or_error = resolver->resolve(socket_address);
+  THROW_IF_STATUS_NOT_OK(instance_or_error, throw);
+  return std::move(instance_or_error.value());
```
Comment (wbpcode) [inline source/common/network/resolver_impl.cc:85]: Should we update the `resolveProtoAddress`  and `resolveProtoSocketAddress` to remove these `throwEnvoyExceptionOrPanic` and throw directly in this PR?

**Review by wbpcode — COMMENTED** [review]
LGTM overall, only one nit comment?

File: source/common/network/resolver_impl.cc (hunk 1 above)
Comment (alyssawilk) [reply source/common/network/resolver_impl.cc:85]: largely I'm trying to tackle one concept at a time - I think I will get to them all (or at least the ones in the Envoy Mobile build, as we have compile checks on those)

**Review by wbpcode — APPROVED** [review]
Thanks.

**Comment by phlax** [comment]
@alyssawilk i think this broke mobile-cc tests - they are permafailing since it landed

**Comment by phlax** [comment]
and mobile-ios, mobile-release-validation, maybe others

**Pull request merged**